src/memory.rs

```diff
@@ -114,3 +121,51 @@ pub fn alloc_cons<'a, 'b, A>(
         OCaml::new(cr, ocaml_cons)
     }
 }
+
+const BOX_OPS_DYN_DROP: custom_operations = custom_operations {
+    identifier: "_rust_box_dyn_drop\0".as_ptr() as *const i8,
+    finalize: Some(drop_box_dyn),
+    compare: None,
+    hash: None,
+    serialize: None,
+    deserialize: None,
+    compare_ext: None,
+    fixed_length: ptr::null(),
+};
+
+extern "C" fn drop_box_dyn(oval: RawOCaml) {
+    unsafe {
+        let box_ptr = ocaml_sys::field(oval, 1) as *mut Pin<Box<dyn Any>>;
+        ptr::drop_in_place(box_ptr);
+    }
+}
+
+/// Implementation notes: is it possible to reduce indirection?
```

Comment:
@g2p can you rebase this one against `master` (merging #20 caused a conflict) and change this comment so that it is not a documentation comment? (what is said here is more about internal implementation details)

Reply:
Actually, nevermind, I am doing it on a new branch where I'm making a few changes and updating the CHANGELOG before merging.